### ISSUE TITLE
borg list without checkpoint archives

### ISSUE THREAD

**Issue by witten** [borgbackup/borg#4788]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yup.

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE / feature request.

#### Your borg version (borg -V).

1.1.10

#### Operating system (distribution) and version.

Manjaro Linux 18.1.1

#### Hardware / network configuration, and filesystems used.

Lenovo X1 Carbon ext4

#### How much data is handled by borg?

Very little in this contrived example: ~164 MB.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

First, interrupt `borg create` with ctrl-C such that a checkpoint archive is created. (It's easier to do this by lowering the checkpoint interval artificially.)

Then:

```
# borg list test.borg
...
flux-2019-10-16T10:00:42.004841      Wed, 2019-10-16 10:00:42 [8f68292443590242781686b9d1c4e1a702903292dce146193a9133e4a5647985]
flux-2019-10-16T10:01:13.113527.checkpoint Wed, 2019-10-16 10:01:13 [d4fceca2de82f9629edbfa151e4ff2ced15cc6b80d81b571a4d3bbe5d4403b7d]
...
```

## Describe the problem you're observing.

It's not a problem so much as a feature request. I'd like to be able to `borg list` while having Borg omit checkpoint archives. My motivation is that I'm trying to programmatically find completed (non-checkpoint) archives, so that I can report on / monitor whether Borg is successfully creating archives.

It's possible to do this checkpoint filtering outside of Borg via `grep` or similar, but as discussed in #4047, that would prevent me from using this checkpoint filtering in conjunction with other `borg list` filters like `--last`, and/or in conjunction with `--json`.

So it would be really great if Borg had an option like `--consider-checkpoints`, similar to the existing `--consider-part-files`. Note that this implies inverting the default behavior, and omitting checkpoints from archive listing by default.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

I can reproduce `borg list` showing checkpoint archives, as that is the default behavior in current versions of Borg.

#### Include any warning/errors/backtraces from the system logs

N/A

**Comment by ThomasWaldmann** [comment]
A good time to introduce this will be borg 1.2 (== master branch now).

**Comment by natea123** [comment]
Eager to try and take a crack at this if its still available.

**Comment by ThomasWaldmann** [comment]
It's still open, so that means it is still todo. So just try and make a PR. :)

**Comment by ThomasWaldmann** [comment]
guess this is solved by #4918.
